experimental/unitsconversion/src/converter.rs

```diff
@@ -0,0 +1,345 @@
+        // Apply power.
+        conversion_info_factor = conversion_info_factor.pow((unit_item.power * sign) as i32);
+        Some(conversion_info_factor)
+    }
+
+    fn extract_ratio_from_unaligned(
+        sign_ule: &SignULE,
+        num_ule: &ZeroSlice<u8>,
+        den_ule: &ZeroSlice<u8>,
+    ) -> Ratio<BigInt> {
+        let sign = match Sign::from_unaligned(*sign_ule) {
+            Sign::Positive => num_bigint::Sign::Plus,
+            Sign::Negative => num_bigint::Sign::Minus,
+        };
+
+        Ratio::<BigInt>::new(
+            BigInt::from_bytes_le(sign, num_ule.as_ule_slice()),
+            BigInt::from_bytes_le(num_bigint::Sign::Plus, den_ule.as_ule_slice()),
+        )
+    }
+
+    fn compute_offset(
+        &self,
+        input_unit: &MeasureUnit,
+        output_unit: &MeasureUnit,
+    ) -> Option<Ratio<BigInt>> {
+        // In order to have an offset, the input and output units should be simple.
+        // This means, the input and output units should have only one unit item with power 1 and si prefix 0.
+        // For example:
+        //           1 - `meter` and `foot` are simple units.
+        //           2 - `meter-per-second` and `foot-per-second` are not simple units.
+        if !(input_unit.contained_units.len() == 1
+            && output_unit.contained_units.len() == 1
+            && input_unit.contained_units[0].power == 1
+            && output_unit.contained_units[0].power == 1
+            && input_unit.contained_units[0].si_prefix.power == 0
+            && output_unit.contained_units[0].si_prefix.power == 0)
+        {
+            return Some(Ratio::<BigInt>::from_integer(0.into()));
+        }
+
+        let input_conversion_info = self
+            .payload
+            .convert_infos
+            .get(input_unit.contained_units[0].unit_id as usize)?;
+
+        let output_conversion_info = self
+            .payload
+            .convert_infos
+            .get(output_unit.contained_units[0].unit_id as usize)?;
+
+        let input_offset = Self::extract_ratio_from_unaligned(
+            &input_conversion_info.offset_sign,
+            input_conversion_info.offset_num(),
+            input_conversion_info.offset_den(),
+        );
+
+        let output_offset = Self::extract_ratio_from_unaligned(
+            &output_conversion_info.offset_sign,
+            output_conversion_info.offset_num(),
+            output_conversion_info.offset_den(),
+        );
+
+        if input_offset.is_zero() && output_offset.is_zero() {
+            return Some(Ratio::<BigInt>::from_integer(0.into()));
+        }
+
+        let output_conversion_rate_recip = Self::extract_ratio_from_unaligned(
+            &output_conversion_info.factor_sign,
+            // Because we are computing the reciprocal, the numerator and denominator are swapped.
+            output_conversion_info.factor_den(),
+            output_conversion_info.factor_num(),
+        );
```

Comment:
Question: why do you need `output_conversion_rate_recip`? An example in a comment could help.

Reply:
This is an equation:

I added a big illustration for it.